andela/northstar-backend
# #168134583 Update database

### What does this PR do?
Updates database to accommodate changes made by the Product Owner.

### Description of tasks to be completed
- Delete the `departments` table
- Delete the `suppliers` table: as it stands, only `travel_admins` can add facilities
- Use `manager_id` in place of `department_id` in the `users` table
- Rename `trips` table to `requests`
- Delete the `travel_team_member` role
- Add a Boolean column `email_notifications` to the users' table and default it to `true` : this column, if set to `false`, will opt users out from email notifications
- Replace `from` with `origin` in requests table
- Add a Boolean column `checked_in` to the `bookings` table and set it to false: this will ensure a user only rates a facility that he/she has checked-in to
- Delete the `replied_to` column on the `comments` table: comments should in a single thread.
- Associate models properly

### How can this be manually tested?
- Download or clone this repository
- On your terminal, cd into the application's directory and run: git checkout `bg-database-168134583`
-  Set the following environment variables in your `.env` file:
```
DEV_DB_USER // database username
DEV_DB_PASSWORD // database password
DEV_DATABASE  //name of your local database
DATABASE_URL  // production database url
```

- If you have not, install [Postgresql](https://www.postgresql.org/download)
- Run `npm run db:migrate`
- Check your database, you should now have the following tables:
  - users
  - facilities
  - likes
  - rooms
  - amenities
  - feedbacks
  - bookings
  - ratings
  - requests
  - comments
  - notifications
- To undo migration, run `npm run undo:migrate`

### What are the relevant Pivotal Tacker stories?
[#168134583 ](https://www.pivotaltracker.com/story/show/168134583 )
